torch/csrc/dynamo/eval_frame.c

```diff
@@ -776,26 +799,37 @@ static PyObject* _custom_eval_frame(
   eval_frame_callback_set(Py_None);
 
   PyObject* hook_record = call_profiler_start_hook(guard_profiler_name_str);
-  PyObject* maybe_cached_code = lookup(extra, frame, NULL, 0);
+  PyObject* code_part = NULL;
+  PyObject* maybe_cached_code = lookup(extra, frame, NULL, 0, &code_part);
+  Py_XINCREF(code_part);
```

Comment:
Doesn't lookup already incref? Why are you incref'ing again?

Reply:
Actually, this is the right one, I think? Because it can be null. let me double check.